core/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoHashGridParser.java

```diff
@@ -141,6 +155,23 @@ public InternalAggregation buildEmptyAggregation() {
         protected Aggregator doCreateInternal(final ValuesSource.GeoPoint valuesSource, AggregationContext aggregationContext,
                 Aggregator parent, boolean collectsFromSingleBucket, List<PipelineAggregator> pipelineAggregators, Map<String, Object> metaData)
                 throws IOException {
+            if (shardSize == 0) {
+                shardSize = Integer.MAX_VALUE;
+            }
+
+            if (requiredSize == 0) {
+                requiredSize = Integer.MAX_VALUE;
```

Comment:
So we have a loose convention that `size: 0` means get all buckets. I personally don't like it but we have to keep it for now I guess as this refactor should be backwards compatible